src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setOrderDate(LocalDateTime.now());
+        order.setTickets(new ArrayList<>(shoppingCart.getTickets()));
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        return orderDao.add(order);
```

Comment:
як на мене це буде дільш правильна поведінка. Якщо станеться помилка orderDao.add(order), то ми втратимо і ті квитки в шопінг карті, бо вони вже були видалені, виходить і ордер не створили і квитки уже видалили
```suggestion
        order.setTickets(new ArrayList<>(shoppingCart.getTickets()));
        orderDao.add(order);
        shoppingCartService.clearShoppingCart(shoppingCart);
        return order;
```

Reply:
Згоден! Я спочатку так і написав) ... але щось потім подумав що треба повернути ордер який повернувся з методу add() тобто вже потрапив у базу. Дякую!